mailbox/client_conn.go

```diff
@@ -157,20 +157,32 @@ func NewClientConn(ctx context.Context, sid [64]byte, serverHost string,
 	}
 
 	c := &ClientConn{
-		transport: transport,
-		gbnOptions: []gbn.Option{
-			gbn.WithTimeout(gbnTimeout),
-			gbn.WithHandshakeTimeout(gbnHandshakeTimeout),
-			gbn.WithKeepalivePing(
-				gbnClientPingTimeout, gbnPongTimeout,
-			),
-		},
+		transport:   transport,
 		status:      ClientStatusNotConnected,
 		onNewStatus: onNewStatus,
 		quit:        make(chan struct{}),
 		cancel:      cancel,
 		log:         logger,
 	}
+
+	c.gbnOptions = []gbn.Option{
+		gbn.WithTimeout(gbnTimeout),
+		gbn.WithHandshakeTimeout(gbnHandshakeTimeout),
+		gbn.WithKeepalivePing(
+			gbnClientPingTimeout, gbnPongTimeout,
+		),
+		gbn.WithOnFIN(func() {
```

Comment:
We could potentially add this to the `server_conn.go` version as well, and force the server to set the connection status to status `ServerStatusIdle` as well. But I haven't really seen any case where this has caused any issues on the server side yet, so opted to not add it. But Will add it to the server side as well if you think I should!.